src/winc1500/src/winc.c

```diff
@@ -820,23 +822,33 @@ int winc_socket_connect(int fd, sockaddr *addr, uint32_t timeout)
 
 int winc_socket_send(int fd, const uint8_t *buf, uint32_t len, uint32_t timeout)
 {
+    uint32_t tick_start = HAL_GetTick();
+
     int bytes = 0;
 
     while (bytes < len) {
         // Split the packet into smaller ones.
         int n = MIN((len - bytes), SOCKET_BUFFER_MAX_LENGTH); 
-        int ret = WINC1500_EXPORT(send)(fd, (uint8_t*)buf + bytes, n, timeout);
+        int ret = WINC1500_EXPORT(send)(fd, (uint8_t*)buf + bytes, n, 0);
 
         if (ret == SOCK_ERR_NO_ERROR) {
             // Do async request
-            ret = winc_async_request(SOCKET_MSG_SEND, &n, timeout);
+            int async_ret;
+            ret = winc_async_request(SOCKET_MSG_SEND, &async_ret, timeout);
 
             // Check sent bytes returned from async request.
-            if (ret != SOCK_ERR_NO_ERROR || n <= 0) {
-                return (n <= 0)? n : ret;
-            }
+            if (ret != SOCK_ERR_NO_ERROR) return ret;
```

Comment:
This is basically the same as the old code. There's no reason to change it (except for n<0), and please reuse n.

Reply:
I can't do this for all cases because sendto returns 0 for n always. For consistency I don't want to do this.